### PR TITLE
Contract demo

### PR DESCRIPTION
**Pull request by torsten-systemticks**

### CODE REVIEW

File: src/com.elektrobit.ebrace.core.interactor/src/com/elektrobit/ebrace/core/interactor/createResource/CreateConnectionInteractionUseCaseImpl.java [hunk 1]
```diff
@@ -1,90 +1,110 @@
...
+        Optional<ConnectionModel> matchingConnection = resourcesModelManager.getConnections().stream()
+                .map( c -> (ConnectionModel)c )
+                .filter( c -> c.getName().equals( name ) && c.getPort() == port && c.getHost().equals( host ) )
+                .findFirst();
+
+        if (matchingConnection.isPresent())
+        {
+            return matchingConnection.get();
+        }
+
```
Comment (raphael-systemticks) [inline src/com.elektrobit.ebrace.core.interactor/src/com/elektrobit/ebrace/core/interactor/createResource/CreateConnectionInteractionUseCaseImpl.java]: Can your remove that empty line between if and else block?

File: src/com.elektrobit.ebrace.racescript.interface/src/com/elektrobit/ebsolys/script/external/ScriptContext.java [hunk 2]
```diff
@@ -1,752 +1,783 @@
...
+    /**
+     * Retrieve an existing connection that can be later used to connect to target.
+     * 
+     * @param name
+     *            Name of connection visible in the UI
+     * @param host
+     *            Host name or IP address of the target
+     * @param port
+     *            Port to connect to at target (port of Target Agent)
+     * @return
```
Comment (raphael-systemticks) [inline src/com.elektrobit.ebrace.racescript.interface/src/com/elektrobit/ebsolys/script/external/ScriptContext.java]: Can you document here that the result can be null if connection wasn't found.

**Comment by basmatix** [comment]
Shouldn't we stick to the createOrGetX philosophy instead of breaking it up in two methods? Or does this not fit the connections use case? In the end we still can add the createOrGetConnection method later...

**Comment by torsten-systemticks** [comment]
Actually we have those kind of methods in the API, e.g.:
createOrGetChannel
getChannel

or

createOrGetChart
getChart

I wanted to apply paradigm the same for the connection handling

**Comment by basmatix** [comment]
Ah, I see! Sorry, I thought, seperate create and get were to replace the createOrGet. But if it is just an addition, it's fine for me.

**Comment by torsten-systemticks** [comment]
No, it is just the other way round: 'createOrGet' was missing. :)
I have set the existing 'create' to deprecated.

File: src/com.elektrobit.ebrace.core.interactor/src/com/elektrobit/ebrace/core/interactor/createResource/CreateConnectionInteractionUseCaseImpl.java (hunk 1 above)
Comment (torsten-systemticks) [reply src/com.elektrobit.ebrace.core.interactor/src/com/elektrobit/ebrace/core/interactor/createResource/CreateConnectionInteractionUseCaseImpl.java]: Done

File: src/com.elektrobit.ebrace.racescript.interface/src/com/elektrobit/ebsolys/script/external/ScriptContext.java (hunk 2 above)
Comment (torsten-systemticks) [reply src/com.elektrobit.ebrace.racescript.interface/src/com/elektrobit/ebsolys/script/external/ScriptContext.java]: Done

**Pull request merged**